### PR TITLE
issue45-iter2

### PR DESCRIPTION
**Pull request by maggienj**
Creating this pull request just to show changes only. this is not an "actual pull request" as this branch contains a lot of "warm-up" codes to understand what changes are required to pass this selected test.

### CODE REVIEW

File: active_data/public/modevlib/Dimension-Platform.js [hunk 1]
```diff
@@ -386,8 +386,8 @@ if (!Mozilla) var Mozilla = {"name": "Mozilla", "edges": []};
                   "esfilter": {"nested": {
                     "path": "attachments.flags",
                     "query": {"bool": {
-                      "query": {"match_all": {}},
-                      "filter": {"and": [
+                      "must": {"match_all": {}},
```
Comment (klahnakoski) [inline active_data/public/modevlib/Dimension-Platform.js]: Old ES demanded that a query has a "query" property (for scoring results) and an optional "filter" property.  Notice the `{"query": {"match_all": {}}}` does nothing.  Remove it.

File: active_data/public/modevlib/Dimension-Platform.js [hunk 2]
```diff
@@ -386,8 +386,8 @@ if (!Mozilla) var Mozilla = {"name": "Mozilla", "edges": []};
...
+                      "filter": {"must": [
```
Comment (klahnakoski) [inline active_data/public/modevlib/Dimension-Platform.js]: Recall that `{"and": []}` must be converted to `{"bool": {"must": []}}`.

File: jx_elasticsearch/es14/decoders.py [hunk 3]
```diff
@@ -170,11 +170,11 @@ def append_query(self, es_query, start):
                         "order": {"_term": self.sorted} if self.sorted else None
                     }}, es_query),
                     "_missing": set_default(
-                        {"filter": OrOp("or", [
-                            field.missing(),
-                            NotOp("not", InOp("in", [Variable(field.var), Literal("literal", include)]))
-                        ]).to_esfilter()},
-                        es_query
+                        {"query": {"bool": {"filter": OrOp("or", [
```
Comment (klahnakoski) [inline jx_elasticsearch/es14/decoders.py]: Do not make this change: The `filter` here is in reference to the filter aggregation, not a query filter.

**Comment by maggienj** [comment]
this could be closed, as a new pull req has been created.

**Pull request closed without merging**